### PR TITLE
Add config object for Griddle

### PR DESCRIPTION
**Pull request by shannynalayna**
This change adds a 'config' module to Griddle (currently unused) that will be utilized to build the Griddle daemon to run. This allows for config values to be collected from various sources, currently including environment vars, default values, and clap arguments. 

Testing: 

There are multiple unit tests for the various configuration objects. You can run the different combinations using the various features. There are three main features that have tests written behind. To run the full suite of Griddle tests, including the configuration tests, run `cargo test --features experimental`. If you'd like to run the individual config tests, run the following: 

```
cargo test --features config-env

cargo test --features config-default 

cargo test --features config-clap
```

Assumptions: 
- Use an enum to express the implementation of `ScopeId`. Griddle only needs to know the implementation of the `ScopeId` trait, as it does not need access to the backend. Griddle also does not need to know any service scope ID on start-up, as it will take in requests for various services. Knowing the impl type of scope id, Griddle can validate that argument in requests, validating that the scope ID argument is valid / well-formatted. Griddle can return an error if it does /not/ expect that argument.  

Also, currently the usage for that scope enum looks like this:

```
griddle -s [service,global] / --scope [service,global] 
```

This could also possibly be represented as a flag, avoiding the need to parse user input for the variant of scope ID used. This could be represented as `griddle --[service / global]`.

### CODE REVIEW

**Review by ameliabradley — APPROVED** [review]

File: griddle/src/main.rs [hunk 1]
```diff
@@ -15,6 +15,9 @@
 #[macro_use]
 extern crate log;
 
+#[allow(dead_code)]
```
Comment (CLEckhardt) [inline griddle/src/main.rs:18]: Should this be removed after this commit?

Comment (shannynalayna) [reply griddle/src/main.rs:18]: Ah yes it should be , good catch

File: griddle/src/config/default.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    /// Verifies that a `PartialGriddleConfig` object is accurately constructed from default values.
+    /// The follows these steps:
```
[the remark this answers is not in the document]
Comment (shannynalayna) [reply griddle/src/config/default.rs]: Fixed

File: griddle/src/config/env.rs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            .expect("Unable to build `EnvPartialGriddleConfigBuilder`");
+        assert_eq!(
+            empty_generated_config.source(),
+            GriddleConfigSource::Environment
+        );
+        // Assert the config object's values are not set.
+        assert_eq!(empty_generated_config.signing_key(), None);
+        assert_eq!(empty_generated_config.rest_api_endpoint(), None);
+
+        // Create a new EnvPartialGriddleConfigBuilder object.
```
[the remark this answers is not in the document]
Comment (shannynalayna) [reply griddle/src/config/env.rs:167]: Separated the "empty" test and the actual value one

**Review by CLEckhardt — APPROVED** [review]

**Review by ameliabradley — APPROVED** [review]

**Pull request merged**